### PR TITLE
Add bean validation for primitive container type 17450

### PR DESCRIPTION
**Pull request by MelleD**
Add Bean validation for primitive type for container see https://github.com/OpenAPITools/openapi-generator/issues/17150

Change should not be breaking.

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@cachescrubber (2022/02) @welshm (2022/02) @MelleD (2022/02) @atextor (2022/02) @manedev79 (2022/02) @javisst (2022/02) @borsch (2022/02) @banlevente (2022/02) @Zomzog (2022/09)

### CODE REVIEW

**Comment by MelleD** [comment]
@cachescrubber (2022/02) @welshm (2022/02) @MelleD (2022/02) @atextor (2022/02) @manedev79 (2022/02) @javisst (2022/02) @borsch (2022/02) @banlevente (2022/02) @Zomzog (2022/09)

any updates here?

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/SpringCodegen.java [hunk 1]
```diff
@@ -1459,16 +1459,85 @@ private static boolean isListOrSet(CodegenParameter codegenParameter) {
...
         if (StringUtils.isEmpty( dataType ) || dataType.contains( "@Valid" )) {
             return dataType;
         }
-        return dataType.replace( "<", "<@Valid " );
+
+        if (codegenProperty.isModel) {
+            return dataType.replace( "<", "<@Valid " );
+        }
+        String beanValidation = getPrimitiveBeanValidation(codegenProperty);
+        if(beanValidation == null) {
```
Comment (welshm) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/SpringCodegen.java:1474]: style nit: inconsistent spacing before `(`

**Review by welshm — APPROVED** [review]

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/SpringCodegen.java (hunk 1 above)
Comment (MelleD) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/SpringCodegen.java:1474]: Iam not sure why this code style is not working with IntelliJ. Do you have hint for me ?

Comment (welshm) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/SpringCodegen.java:1474]: Ah no I don't have any hints sorry 😔 

the formatting is not entirely consistent for this project either. I usually just try to match spacing and new line styles to the code immediately above and below myself.

Not a big deal either way

**Comment by MelleD** [comment]
@wing328 fyi

**Pull request merged**